File: src/test/java/model/PerformanceTest.java
package model;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.time.LocalDate;

import static org.junit.jupiter.api.Assertions.*;

class PerformanceTest {
    Performance performance;
    Performance performance1;

    @BeforeEach
    void setUp() {
        performance = new Performance(LocalDate.now(), "Cluj_Napoca", 206, 1678, "Kygo");
        performance = new Performance(1L, LocalDate.now(), "Cluj-Napoca", 206, 1678, "Kygo");
    }

    @Test
    void getDate() {
        assertEquals(performance.getDate(), LocalDate.now());
    }

    @Test
    void setDate() {
        performance.setDate(LocalDate.of(2022, 8, 26));
        assertEquals(performance.getDate(), LocalDate.of(2022, 8, 26));
    }

    @Test
    void getPlace() {
        assertEquals(performance.getPlace(), "Cluj-Napoca");
    }

    @Test
    void setPlace() {
        performance.setPlace("Bontida");
        assertEquals(performance.getPlace(), "Bontida");
    }

    @Test
    void getNoOfAvailableTickets() {
        assertEquals(performance.getNoOfAvailableSeats(), 206);
    }

    @Test
    void setNoOfAvailableTickets() {
        performance.setNoOfAvailableSeats(205);
        assertEquals(performance.getNoOfAvailableSeats(), 205);
    }

    @Test
    void getNoOfSoldTickets() {
        assertEquals(performance.getNoOfSoldSeats(), 1678);
    }

    @Test
    void setNoOfSoldTickets() {
        performance.setNoOfSoldSeats(1679);
        assertEquals(performance.getNoOfSoldSeats(), 1679);
    }

    @Test
    void getArtists() {
        assertEquals(performance.getArtist(), "Kygo");
    }

    @Test
    void setArtists() {
        performance.setArtists("21P");
        assertEquals(performance.getArtist(), "21P");
    }
}

File: Tests/model/PerformanceTest.cs
using System;
using System.Collections.Generic;
using MusicFestivalMPP.model;
using NUnit.Framework;

namespace Tests.model;

public class PerformanceTest
{
    private Performance _performance;
    private Performance _performance0; 
    
    [SetUp]
    public void Setup()
    {
        DateTime date = new DateTime(2022, 8, 26);
        _performance = new Performance(1, new DateTime(2022, 8, 26), "CJ", 100, 200, "21P");
        _performance0 = new Performance(date, "CJ", 100, 200, "21P");
    }

    [Test]
    public void GetDate()
    {
        Assert.AreEqual(_performance.Date, new DateTime(2022, 8, 26));
        Assert.AreEqual(_performance0.Date, new DateTime(2022, 8, 26));
    }
    
    [Test]
    public void SetDate()
    {
        _performance.Date = new DateTime(2022, 3, 22);
        Assert.AreEqual(_performance.Date, new DateTime(2022, 3, 22));
        Assert.AreEqual(_performance0.Date, new DateTime(2022, 8, 26));
    }
    
    [Test]
    public void GetPlace()
    {
        Assert.AreEqual(_performance.Place, "CJ");
    }
    
    [Test]
    public void SetPlace()
    {
        _performance.Place = "Cluj-Napoca";
        Assert.AreEqual(_performance.Place, "Cluj-Napoca");
    }
    
    [Test]
    public void GetAvailable()
    {
        Assert.AreEqual(_performance.NoOfAvailableSeats, 100);
    }
    
    [Test]
    public void SetAvailable()
    {
        _performance.NoOfAvailableSeats = 105;
        Assert.AreEqual(_performance.NoOfAvailableSeats, 105);
    }
    
    [Test]
    public void GetSold()
    {
        Assert.AreEqual(_performance.NoOfSoldSeats, 200);
    }
    
    [Test]
    public void SetSold()
    {
        _performance.NoOfSoldSeats = 195;
        Assert.AreEqual(_performance.NoOfSoldSeats, 195);
    }
    
    [Test]
    public void GetArtists()
    {
        Assert.AreEqual(_performance.Artist, "21P");
    }
    
    [Test]
    public void SetArtists()
    {
        _performance.Artist = "Kygo";
        Assert.AreEqual(_performance.Artist, "Kygo");
    }
}
